docs/topics/coding-conventions.md

```diff
@@ -74,19 +76,21 @@ root
 ```
 
 Here both `Platform.kt` files are in the same package, so the Kotlin JVM compiler produces two file facades, both of which
-have FQN `myPackage.PlatformKt`. Starting with Kotlin 1.8.20, this produces the "Duplicate JVM classes" error.
+have FQN `myPackage.PlatformKt`. This produces the "Duplicate JVM classes" error.
 
 The simplest way to avoid that is renaming one of the files according to the guideline above. This naming scheme helps
-avoid such clashes while retaining code readability.
+avoid clashes while retaining code readability.
 
-> * Non-JVM platforms are not prone to this issue. However, following these recommendations can help you keep file naming
-> consistent throughout the project.
-> * On JVM, you can have duplicate paths in common and platform source sets if source files do not have top-level
-> declarations because the file facades aren't generated in this case.
+> There are two cases when these recommendations may seem redundant, but we still advise to follow them:
+> 
+> * Non-JVM platforms don't have issues with duplicating file facades. However, this naming scheme can help you keep
+> file naming consistent.
+> * On JVM, if source files don't have top-level declarations, the file facades aren't generated, and you won't face
+> naming clashes.
+> 
+>   However, this naming scheme can help you avoid issues in the future. A simple refactoring
+> or an addition that ends up in a top-level function will lead to the same "Duplicate JVM classes" error.
```

Comment:
It's unclear whether this sentence refers to the situation without guidelines applied, or with guideline applied (note that by default, e.g. while scanning the whole page, I'd personally most likely assume the latter, which is incorrect). Please clarify it, e.g.:

```
However, this naming scheme can help you avoid situations when a simple refactoring or an addition adds a top-level member and provoke the same "Duplicate JVM classes" error
```